core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -451,4 +519,26 @@ private void deleteDataFiles(Set<ManifestFile> manifestsToScan, Set<ManifestFile
       return CloseableIterable.withNoopClose(snapshot.allManifests());
     }
   }
+
+  private class SnapshotTimestampComparator implements Comparator<Snapshot> {
+
+    @Override
+    public int compare(Snapshot o1, Snapshot o2) {
+      if (o1.timestampMillis() < o2.timestampMillis()) {
+        return -1;
+      }
+      else if (o1.timestampMillis() > o2.timestampMillis()) {
+        return 1;
+      }
+      else {
+        if (o1.snapshotId() > o2.snapshotId()) {
+          return -1;
+        }
+        else if (o1.snapshotId() < o2.snapshotId()) {
+          return 1;
+        }
+        return 0;
+      }
+    }
```

Comment:
If we end up doing this, I think this should be reversed. The latest snapshot should be at the top of the heap.